conans/client/conan_api.py

```diff
@@ -1165,20 +1202,99 @@ def editable_remove(self, reference):
     def editable_list(self):
         return {str(k): v for k, v in self._cache.editable_packages.edited_refs.items()}
 
+    @api_method
+    def update_lock(self, old_lockfile, new_lockfile, cwd=None):
+        cwd = cwd or os.getcwd()
+        old_lockfile = _make_abs_path(old_lockfile, cwd)
+        old_lock = GraphLockFile.load(old_lockfile)
+        new_lockfile = _make_abs_path(new_lockfile, cwd)
+        new_lock = GraphLockFile.load(new_lockfile)
+        if old_lock.profile.dumps() != new_lock.profile.dumps():
+            raise ConanException("Profiles of lockfiles are different\n%s\n%s"
```

Comment:
I had this pending from the previous review (I forgot to submit).

```
This is printing:

ERROR: Profiles of lockfiles are different
<conans.model.profile.Profile object at 0x7fd112338e10>
<conans.model.profile.Profile object at 0x7fd112346470>
```

I think this is still happening.

Reply:
I am fixing the output, but leaving the error. If this happens is because the profiles are different, we don't want that, do we?

Reply:
Yes yes, I was only suggesting to fix the output of the profile as "object"